src/main.rs

```diff
@@ -63,1116 +66,895 @@ fn main() -> Result<()> {
+    Tail {
+        /// Specify an output format
+        #[structopt(long, short = "f", default_value = "json", possible_values = &["json", "pretty"])]
+        format: String,
+
+        /// Port to accept tail log requests
+        #[structopt(long = "port", short = "p")]
+        tunnel_port: Option<u16>,
+
+        /// Provides endpoint for cloudflared metrics. Used to retrieve tunnel url
+        #[structopt(long = "metrics")]
+        metrics_port: Option<u16>,
+    },
+
+    /// Authenticate Wrangler with your Cloudflare username and password
+    #[structopt(name = "login")]
+    Login,
+
+    /// Report an error caught by wrangler to Cloudflare
+    #[structopt(name = "report")]
+    Report {
+        /// Specifies a log to report (e.g. --log=1619728882567.log)
+        #[structopt(name = "log", long)]
+        log: Option<PathBuf>,
+    },
+}
 
-        let verify = !matches.is_present("no-verify");
+#[derive(Debug, StructOpt)]
+pub struct AdhocMigration {
+    /// Allow durable objects to be created from a class in your script
+    #[structopt(name = "new-class", long, number_of_values = 1)]
+    new_class: Vec<String>,
 
-        commands::global_config(&user, verify)?;
-    } else if let Some(matches) = matches.subcommand_matches("generate") {
-        let name = matches.value_of("name").unwrap_or("worker");
-        let site = matches.is_present("site");
-        let template = matches.value_of("template");
-        let mut target_type = None;
+    /// Delete all durable objects associated with a class in your script
+    #[structopt(name = "delete-class", long, number_of_values = 1)]
+    delete_class: Vec<String>,
 
-        let template = if site {
-            if template.is_some() {
-                anyhow::bail!("You cannot pass a template and the --site flag to wrangler generate. If you'd like to use the default site boilerplate, run wrangler generate --site. If you'd like to use another site boilerplate, omit --site when running wrangler generate.")
-            }
-            "https://github.com/cloudflare/worker-sites-template"
-        } else {
-            if let Some(type_value) = matches.value_of("type") {
-                target_type = Some(TargetType::from_str(&type_value.to_lowercase())?);
-            }
+    /// Rename a durable object class
+    #[structopt(name = "rename-class", long, number_of_values = 2)]
+    rename_class: Vec<String>,
 
-            let default_template = "https://github.com/cloudflare/worker-template";
-            template.unwrap_or(match target_type {
-                Some(ref pt) => match pt {
-                    TargetType::Rust => "https://github.com/cloudflare/rustwasm-worker-template",
-                    _ => default_template,
+    /// Transfer all durable objects associated with a class in another script to a class in
+    /// this script
+    #[structopt(name = "transfer-class", long, number_of_values = 3)]
+    transfer_class: Vec<String>,
+}
+
+impl AdhocMigration {
```

Comment:
Is it necessary for this to be in `main.rs` ?

Reply:
Yes (it's processing command line input so I'd argue it belongs in main)

Reply:
ok, I see. good to merge, I may do a little cleanup to get more of this out of main.